### PR TITLE
fix(js/plugins/ollama): reconfigure ollama embedding

### PR DESCRIPTION
**Pull request by cabljac**
Should resolve https://github.com/firebase/genkit/issues/1080

Also enabled the tests for this plugin.

Checklist (if applicable):
- [x] Tested (manually, unit tested, etc.)
- [x] Docs updated

### CODE REVIEW

File: docs/plugins/ollama.md [hunk 1]
```diff
@@ -122,3 +122,48 @@ const llmResponse = await generate({
...
+
+```js
+import { ollama } from 'genkitx-ollama';
+
+export default configureGenkit({
+  plugins: [
+    ollama({
+      embedders: [
+        {
+          name: 'nomic-embed-text', // specify the embedding model
```
[the remark this answers is not in the document]
Comment (cabljac) [reply docs/plugins/ollama.md:141]: yup, good catch

File: js/plugins/ollama/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  serverAddress: string;
+
+  /**
+   * Optional request headers, which can be either static or dynamically generated.
+   */
+  requestHeaders?: RequestHeaders;
+}
+
+// Function type for generating request headers dynamically
+export interface RequestHeaderFunction {
```
Comment (pavelgj) [inline js/plugins/ollama/src/types.ts:75]: This looks good, but this is a breaking change from:
```ts
((
  params: { serverAddress: string; model: ModelDefinition },
  request: GenerateRequest
) => Promise<Record<string, string> | void>)
```

I mean, if you look at docs/plugins/ollama.md "Authentication" chapter... then this:

```ts
const ollamaProd = {
  ...ollamaCommon,
  serverAddress: 'https://my-deployment',
  requestHeaders: async (params) => {
    const headers = await fetchWithAuthHeader(params.serverAddress);
    return { Authorization: headers['Authorization'] };
  },
} as OllamaPluginParams;
```

must be updated to 

```ts
const ollamaProd = {
  ...ollamaCommon,
  serverAddress: 'https://my-deployment',
  requestHeaders: async (options) => {
    const headers = await fetchWithAuthHeader(options.params.serverAddress);
    return { Authorization: headers['Authorization'] };
  },
} as OllamaPluginParams;
```

I would prefer to do it without breaking changes (at least to what we have documented)... can we do something like this?

```ts
export interface RequestHeaderFunction {
  (params: {
    serverAddress: string;
    model: ModelDefinition | EmbeddingModelDefinition;
    modelRequest?: GenerateRequest;
    embedRequest?: {docs: Document[], config?: any};
  },
  // @deprecated -- moved into params, here for backwards compatibility reasons.
  modelRequest?: GenerateRequest): Promise<Record<string, string> | void>;
}
```

Comment (cabljac) [reply js/plugins/ollama/src/types.ts:75]: Sure!

**Comment by cabljac** [comment]
closing in favour of PR to `next` [here](https://github.com/firebase/genkit/pull/1246)

**Pull request closed without merging**